packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -196,6 +209,63 @@ export function parseLanguage(className: string): string | undefined {
   return languageClassName?.replace(/language-/, '');
 }
 
+/**
+ * The options configuring the line parsing behavior.
+ */
+export type CodeBlockParseLinesOptions = {
+  /**
+   * The full metastring, as received from MDX. Line ranges declared here
+   * start at 1.
+   */
+  metastring: string | undefined;
+  /**
+   * Language of the code block, used to determine which kinds of magic
+   * comment styles to enable.
+   */
+  language: string | undefined;
+  /**
+   * Magic comment types that we should try to parse. Each entry would
+   * correspond to one class name to apply to each line.
+   */
+  magicComments: MagicCommentConfig[];
+};
+
+/**
+ * The highlighted lines, 0-indexed. e.g. `{ 0: ["highlight", "sample"] }`
+ * means the 1st line should have `highlight` and `sample` as class names.
+ */
+export type CodeBlockLineClassNames = {[lineIndex: number]: string[]};
+
+/**
+ * The supported types for {@link CodeBlockMetaOptions} values.
+ */
+export type CodeMetaOptionValue = string | number | boolean | undefined;
+
+/**
+ * A property bag for custom options specified by the user via metastring
+ * to control aspects like title and line numbers.
+ */
+export type CodeBlockMetaOptions = {
+  [key: string]: CodeMetaOptionValue;
+};
```

Comment:
This is not typesafe, we'd rather use an interface as explained here https://github.com/facebook/docusaurus/pull/11011#discussion_r2005524386